src/System/Process/Typed.hs

```diff
@@ -620,7 +620,8 @@ startProcess pConfig'@ProcessConfig {..} = liftIO $ do
                 -- then call waitForProcess ourselves
                 Left _ -> do
                     P.terminateProcess pHandle
-                    void $ P.waitForProcess pHandle
+                    ec <- P.waitForProcess pHandle
+                    atomically $ putTMVar pExitCode ec
```

Comment:
Two things:

1. Can you update the changelog/do a patch version bump in the cabal file?
2. What do you think about using `tryPutTMVar` instead and using `assert` to ensure that the result value is `True`.